### ISSUE TITLE
Have our own Field trait?

### ISSUE THREAD

**Issue by mimoo** [zksecurity/noname#221]
I'm not sure anymore why we rely on arkworks here. Arkworks comes with a complicated number of traits and a lack of support for serde. In addition different backends might use different versions of arkworks (we've had that problem).

IMO we should have our own Field trait that has a minimal definition, and then we can convert to whatever a backend uses at the right time in the compilation process.

**Comment by katat** [comment]
Agreed, we could just add a number of interfaces to the existing `BackendField` trait.
